master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md

```diff
@@ -22,25 +25,80 @@ You must have a cluster which meets the following requirements:
 - Your Kubernetes controller manager is configured to allocate pod CIDRs (i.e. by passing `--allocate-node-cidrs=true` to the controller manager)
 - Your Kubernetes controller manager has been provided a cluster-cidr (i.e. by passing `--cluster-cidr=192.168.0.0/16`, which the manifest expects by default).
 
+> Note:  If you are upgrading from Calico v2.1, the cluster-cidr selected for your controller manager should remain
+> unchanged from the v2.1 install (the v2.1 manifests default to `10.244.0.0/16`).
+
 ## Installation
 
-To install Calico, ensure you have a cluster which meets the above requirements and run one of the following commands based on your Kubernetes version:
+This document describes three installation options for the two operational modes
+for Calico (Calico with three installation options:
+
+-  Calico policy with Calico networking
+-  Calico policy-only
+   -  with user-supplied networking
+   -  with flannel networking
+
+Ensure you have a cluster which meets the above requirements.  There may be additional requirements based on the installation option you choose.
+
+> Note:  There is currently no upgrade path to switch between different installation options.  Therefore, 
+> if you are upgrading from Calico v2.1, use the [Calico in policy-only mode](#calico-in-policy-only-mode) installation instructions 
+> to upgrade Calico policy-only leaving the networking solution unchanged.
+
+### Calico policy with Calico networking (Beta)
+
+With Kubernetes as the Calico datastore, Calico has Beta support for Calico networking.  This provides BGP-based
+networking with a full node-to-node mesh.  It is not currently possible to configure the Calico BGP network to peer with
+other routers - a future release will provide this feature, and to provide feature parity with the [Kubeadm Hosted Install](../kubeadm)
```

Comment:
This sentence doesn't flow well for me.  What about something like `It is not currently possible to configure the Calico BGP network to peer with other routers - future releases of Calico are expected to bring feature parity with the etcd backed Calico.`